### ISSUE TITLE
`count(): Recursion detected` warning triggered

### ISSUE THREAD

**Issue by sebastianbergmann** [sebastianbergmann/exporter#61]
@staabm I just tagged `6.1.0` with your most recent changes. However, one test from PHPUnit's test suite fails now because a PHP warning is triggered:

```
1) /usr/local/src/phpunit/tests/end-to-end/regression/5614.phpt
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
-PHPUnit %s by Sebastian Bergmann and contributors.
+Warning: count(): Recursion detected in /usr/local/src/phpunit/vendor/sebastian/exporter/src/Exporter.php on line 79
+PHPUnit 11.2.2-15-g3458dc7617 by Sebastian Bergmann and contributors.
 
-Runtime: %s
+Runtime:       PHP 8.3.8
 
+
+Warning: count(): Recursion detected in /usr/local/src/phpunit/vendor/sebastian/exporter/src/Exporter.php on line 79
 .                                                                   1 / 1 (100%)
 
-Time: %s, Memory: %s MB
+Time: 00:00.002, Memory: 8.00 MB
 
 OK (1 test, 1 assertion)
```

This is the test code in question:

```php
final class Issue5614Test extends TestCase
{
    public static function provideRecursiveArray(): iterable
    {
        $array    = [];
        $array[0] = &$array;

        yield [$array];
    }

    #[DataProvider('provideRecursiveArray')]
    public function testRecursiveArray(array $array): void
    {
        $this->assertTrue(true);
    }
}
```

Would suppressing the warning (`@count($data, COUNT_RECURSIVE)` instead of `count($data, COUNT_RECURSIVE)`) be the right thing to do here?

**Comment by staabm** [comment]
> Would suppressing the warning (`@count($data, COUNT_RECURSIVE)` instead of `count($data, COUNT_RECURSIVE)`) be the right thing to do here?

I think you are right, this should work

**Comment by sebastianbergmann** [comment]
Released as `6.1.1`, thank you!
